pkg/datasources/secrets_acceptance_test.go

```diff
@@ -0,0 +1,305 @@
+package datasources_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/assert"
+	accConfig "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/config"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/config/model"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-testing/config"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+const (
+	secretWithClientCredentials      = "snowflake_secret_with_client_credentials"
+	secretWithAuthorizationCodeGrant = "snowflake_secret_with_authorization_code_grant"
+	secretWithBasicAuthentication    = "snowflake_secret_with_basic_authentication"
+	secretWithGenericString          = "snowflake_secret_with_generic_string"
+)
+
+func TestAcc_Secrets_MultipleTypes(t *testing.T) {
```

Comment:
Do we benefit from this test? There are other tests that check single type, we can validate the output thoroughly there, here it would be nice to focus on filtering instead checking the output

Reply:
ok i agree, we would benefit much more by testing filtering on multiple types, and tests each type output independently in tests below